### PR TITLE
Cleanup unused imports

### PR DESCRIPTION
**Pull request by tusharsadhwani**
I wanted to build an [unused import cleaner](https://github.com/tusharsadhwani/zigimports) for Zig, and I ended up using this project as the test-bed for cleaning unused imports and finding edge cases in my implementation.

I'm not sure if such patches are welcome, but I'll just leave a PR here still.

All the test still pass after this changeset, so this shouldn't break anything :)

### CODE REVIEW

File: src/runtime/RuntimeError.zig [hunk 1]
```diff
@@ -11,10 +11,7 @@ const std = @import("std");
 const Allocator = std.mem.Allocator;
 
 const context = @import("context.zig");
-const Activation = @import("./Activation.zig");
 const SourceRange = @import("./SourceRange.zig");
-const VirtualMachine = @import("./VirtualMachine.zig");
-
```
Comment (sin-ack) [inline src/runtime/RuntimeError.zig]: Could you please make it so that the single empty line between imports and the fields are preserved?

File: src/runtime/map_builder.zig [hunk 2]
```diff
@@ -9,8 +9,6 @@ const Slot = @import("slot.zig").Slot;
 const Value = @import("value.zig").Value;
 const AstGen = @import("bytecode/AstGen.zig");
 const context = @import("context.zig");
-const VirtualMachine = @import("VirtualMachine.zig");
-
```
Comment (sin-ack) [inline src/runtime/map_builder.zig]: Same here.

**Review by sin-ack — CHANGES_REQUESTED** [review]
Hey, thanks for this! Your tool looks really nice, I'm thinking of making it a pre-commit linter + CI check. One minor issue I spotted is that it removes empty lines after deleted imports aggressively. I think it should only remove a blank line after an import if there are no more imports left (i.e. removing the lines results in double empty lines); otherwise it should preserve the empty lines.

File: src/runtime/RuntimeError.zig (hunk 1 above)
Comment (tusharsadhwani) [reply src/runtime/RuntimeError.zig]: Right, I can do that.

**Comment by tusharsadhwani** [comment]
@sin-ack that has been taken care of, plus now the tool handles all unused global declarations, not just imports.

**Review by sin-ack — APPROVED** [review]
Thanks, very cool!
